File: backend/app/models/bank.py
from datetime import datetime

from sqlalchemy import Boolean, DateTime, String, func
from sqlalchemy.orm import Mapped, mapped_column, relationship

from app.database import Base


class Bank(Base):
    __tablename__ = "banks"

    id: Mapped[int] = mapped_column(primary_key=True)
    name: Mapped[str] = mapped_column(String(100), nullable=False)
    short_name: Mapped[str] = mapped_column(String(6), nullable=False, index=True)
    is_system: Mapped[bool] = mapped_column(Boolean, default=False, nullable=False)
    created_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True), server_default=func.now(), nullable=False
    )

    accounts = relationship("Account", back_populates="bank")

File: backend/app/models/bank.rb
# == Schema Information
#
# Table name: banks
#
#  id         :bigint           not null, primary key
#  is_system  :boolean          default(FALSE), not null
#  name       :string(100)      not null
#  short_name :string(6)        not null
#  created_at :datetime         not null
#  updated_at :datetime         not null
#
# Indexes
#
#  index_banks_on_short_name  (short_name) UNIQUE
#
class Bank < ApplicationRecord
  has_many :accounts, dependent: :restrict_with_error

  validates :name,       presence: true
  validates :short_name, presence: true, uniqueness: true, length: { maximum: 6 }
end
